server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/EmbeddedCodeNode.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.model.tree;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.Token;
+import org.antlr.v4.runtime.TokenStream;
+import org.antlr.v4.runtime.tree.ParseTreeVisitor;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.semantics.PredefinedVariableContext;
+import org.eclipse.lsp.cobol.core.visitor.Db2SqlVisitor;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.eclipse.lsp.cobol.core.model.tree.NodeType.EMBEDDED_CODE;
+
+/** This class represents embedded code parts in COBOL program, e.g. EXEC CICS and EXEC SQL */
+@Getter
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
```

Comment:
It's not needed to overload equals and hashCode here. Implementation in Node class will be enough.

Reply:
It gives a SonarLint warning if not specified.

Reply:
It is not good idea to use equlas/hashCode from parent, even though it full-fill needs. Each child has to has its own equals/hashCode.

Reply:
Usually, it's not a good idea. But here the parent has a unique ID. For equals check it's enough to check that ID.
I don't know... All that `@EqualsAndHashCode` for nodes looks dirty and needless. Creating a separate predicate for checking that nodes are equal not feels good too.